src/coreclr/vm/jitinterface.cpp

```diff
@@ -6159,7 +6160,8 @@ CORINFO_VARARGS_HANDLE CEEInfo::getVarArgsHandle(CORINFO_SIG_INFO *sig,
     Instantiation methodInst = Instantiation((TypeHandle*) sig->sigInst.methInst, sig->sigInst.methInstCount);
     SigTypeContext typeContext = SigTypeContext(classInst, methodInst);
 
-    result = CORINFO_VARARGS_HANDLE(module->GetVASigCookie(Signature(sig->pSig, sig->cbSig), &typeContext));
+    MethodDesc* pMD = GetMethod(methHnd);
+    result = CORINFO_VARARGS_HANDLE(module->GetVASigCookie(Signature(sig->pSig, sig->cbSig), pMD, &typeContext));
```

Comment:
```suggestion
    result = CORINFO_VARARGS_HANDLE(module->GetVASigCookie(Signature(sig->pSig, sig->cbSig), pMD->IsPInvoke() ? pMD : NULL, &typeContext));
```
I think we need to include the methoddesc in the VASigCookie key only if it is a PInvoke. Ordinary varargs should not need it.